### PR TITLE
aide edit checks and remediations

### PR DESCRIPTION
**Pull request by cyarbrough76**
#### Description:

- _edit oval / bash for periodic cron checking, scan notification and use FIPS hashes. Create cron jobs in accordance with STIG recomendations._

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule aide_periodic_cron_checking:
 Found change in bash remediation.
Rule aide_scan_notification:
 Found change in bash remediation.
Rule aide_use_fips_hashes:
 Text changed in OVAL check.

**Recommended tests to execute:**
 build_product fedora
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-fedora-ds.xml aide_periodic_cron_checking
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml aide_scan_notification
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml aide_use_fips_hashes

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_use_fips_hashes/oval/shared.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
   version="1">
     <ind:filepath>/etc/aide.conf</ind:filepath>
     <ind:pattern operation="pattern match">^[A-Z]*[\s]*=[\s]*.*(sha1|rmd160|sha256|whirlpool|tiger|haval|gost|crc32).*$</ind:pattern>
-    <ind:instance datatype="int" operation="greater than or equal">0</ind:instance>
+    <ind:instance datatype="int" operation="greater than or equal">1</ind:instance>
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/software-integrity/aide/aide_use_fips_hashes/oval/shared.xml:30]: Why this change?

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh [hunk 2]
```diff
@@ -2,9 +2,13 @@
...
-    echo "05 4 * * * root /usr/sbin/aide --check" >> /etc/crontab
+CRONTAB=/etc/crontab
+CRONDIRS='/etc/cron.d /etc/cron.daily /etc/cron.weekly /etc/cron.monthly'
+
+if [ -f /var/spool/cron/root ]; then
+	VARSPOOL=/var/spool/cron/root
+fi
+
+if ! grep -qR '^.*\/usr\/sbin\/aide\s*\-\-check.*|.*\/bin\/mail\s*-s\s*".*"\s*root@.*$' $CRONTAB $VARSPOOL $CRONDIRS; then
+	echo '/usr/sbin/aide --check | /bin/mail -s "$HOSTNAME - Daily aide integrity check run" root@localhost' > /etc/cron.daily/aide
```
Comment (jan-cerny) [inline linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh:13]: I think that mail notifications are already covered by the other rule (aide_scan_notification).

All artefacts within a rule such as OVAL definition, prose in rule.yml, Bash, Ansible, test scenarios and OCIL must but consistent. Over long term, we are trying to eradicate inconsistencies. This change will cause that the text in rule.yml says different thing that what the bash script does. Can you take a look at it?

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_use_fips_hashes/oval/shared.xml (hunk 1 above)
Comment (yuumasato) [reply linux_os/guide/system/software/integrity/software-integrity/aide/aide_use_fips_hashes/oval/shared.xml:30]: This should be fine, the instance count starts at `1`.
https://github.com/OVALProject/Language/blob/master/docs/independent-definitions-schema.md#-textfilecontent54_object-

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh (hunk 2 above)
Comment (redhatrises) [reply linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh:13]: Also, at a minimum aide should run daily, at a maximum, weekly.

Comment (thenefield) [reply linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh:13]: The problem is that we had two different rules to ensure a check was performed, where one indicated a notification and the other didn't.

That resulted in two scan aide checks being performed daily. These aide checks are exhuastive, especially when you have McAfee VirusScan running in the background.

With the proposed change, only one daily check is performed and includes the notification, which in turn will satisfy both rules.

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_use_fips_hashes/oval/shared.xml (hunk 1 above)
Comment (matejak) [reply linux_os/guide/system/software/integrity/software-integrity/aide/aide_use_fips_hashes/oval/shared.xml:30]: This has been already implemented in another PR.

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh (hunk 2 above)
Comment (matejak) [reply linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh:13]: Do we have an answer to this? I don't understand this area, but it looks like a legitimate concern.

**Comment by redhatrises** [comment]
Closing this as the requested changes haven't been made.

**Comment by matejak** [comment]
There is an unanswered question: https://github.com/ComplianceAsCode/content/pull/5903#discussion_r452177778
